File: src/main/scala/uk/co/automatictester/wiremock/maven/plugin/mojo/WireMockStopMojo.scala
package uk.co.automatictester.wiremock.maven.plugin.mojo

import org.apache.maven.plugins.annotations.{LifecyclePhase, Mojo}
import uk.co.automatictester.wiremock.maven.plugin.server.WireMockServer

@Mojo(name = "stop", defaultPhase = LifecyclePhase.POST_INTEGRATION_TEST)
class WireMockStopMojo extends ConfigurationMojo {

    @Override
    def execute: Unit = {
        getLog.info("Stopping WireMock...")
        WireMockServer.stop()
    }
}

File: src/main/java/uk/co/automatictester/wiremock/maven/plugin/mojo/WireMockStopMojo.java
package uk.co.automatictester.wiremock.maven.plugin.mojo;

import org.apache.maven.plugins.annotations.LifecyclePhase;
import org.apache.maven.plugins.annotations.Mojo;
import uk.co.automatictester.wiremock.maven.plugin.server.WireMockServer;

@Mojo(name = "stop", defaultPhase = LifecyclePhase.POST_INTEGRATION_TEST)
public class WireMockStopMojo extends ConfigurationMojo {

    @Override
    public void execute() {
        getLog().info("Stopping WireMock...");
        WireMockServer wireMock = WireMockServer.getInstance();
        wireMock.stop();
    }
}
